paraatm/simulation_method/aviationr.py

```diff
@@ -0,0 +1,217 @@
+            self.simulationInterface.resume(delay)
+            while True:
+                runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                if (runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                    break
+                if (runtime_sim_status == self.NATS_SIMULATION_STATUS_ENDED):
+                    print('Accident condition is not met!')
+                    return None
+            aclist = self.aircraftInterface.getAllAircraftId()
+            ac = self.aircraftInterface.select_aircraft(aclist[0])
+            phase = ac.getFlight_phase()
+            if accident_phase in list(ntsb2nats.loc[ntsb2nats['NATS_CODE'] == phase]['NTSB_CODE']): # test if the phase of flight from  NASTS simulation is same as NTSB
+                accident = True
+        # Start accident simulation and risk analysis
+        risk_list = [None] * len(case)
+        subject_list = [None] * len(case)
+        phase_list = [None] * len(case)
+        occurrence_list = [None] * len(case)
+        time_list = [None] * len(case)
+        risk_estimator = RiskEstimator(self.model, isRNN, self.data, device)
+        
+        model, hierarchical_softmax, risk_model = risk_estimator.load_model()
+        for i in range(1, len(case_code['Occurrence_Code'][0]) - 1):
+
+            # recording the risk and predicting events
+            risk_list[i - 1], phase_list[i - 1], occurrence_list[i - 1], subject_list[i - 1] = risk_estimator.risk_estimation(
+                case_code, i, model, hierarchical_softmax, risk_model, datadict, isRNN)
+
+            time_list[i-1] = self.simulationInterface.get_curr_sim_time()
+
+            # modify NATS simulation according to NTSB accident
+            t = self.accident_simulator(self.aircraftInterface, self.controllerInterface, self.pilotInterface, phase_list[i-1],
+                                        occurrence_list[i-1], subject_list[i-1], datadict, ntsb2nats)
+            self.simulationInterface.resume(t)
+            while True:
+                runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                if (runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                    break
+                if (runtime_sim_status == self.NATS_SIMULATION_STATUS_ENDED):
+                    print('Simulation time not enough!')
+                    return None
+        self.simulationInterface.resume()
+        while True:
+            server_runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+            if (server_runtime_sim_status == self.NATS_SIMULATION_STATUS_ENDED):
+                break
+        event_list = []
+        for i in range(len(subject_list)):
+            event_list.append(phase_list[i] + '\n' + occurrence_list[i] + '\n' + subject_list[i])
+        print('Aviation accident simulation finished!')
+        self.simulationInterface.write_trajectories(self.data + 'trajectory.csv')
```

Comment:
There are a couple of issues with this:

1. It would be better not to call `write_trajectories` in your simulation method.  You are hard-coding the output file name and always producing an output file, even when the user might not want/need it.  I think it would be better just to remove this, and then user can simply call the `write_output` method when they want to write the output.
2. The second issue is that when running `aviationr_example.py`, this write call overwrites `trajectory.csv`, which is a version controlled file.  And actually, I don't think you are using `trajectory.csv` for anything other than writing to it, in which case it should be removed from version control.  I'll make a separate note about this.

`NatsSimulationWrapper.__call__` can actually help you address this, because it will automatically call `write_output` using a temporary file, read the results back into a DataFrame, and return them back to you as a return value of `__call__`.  This is really the intended usage of the simulation wrapper.  The one quirk is that in your case you are returning results from your `simulation` method.  I could update `NatsSimulationWrapper.__call__` to check for this and include those in its results, but that gets a little complicated due to handling several different possible combinations of return types.  If I did make this change, what you could do is:
```python
track, result = sim()
```
Where `sim()` is shorthand for `sim.__call__()`.  That way `NatsSimulationWrapper.__call__` is handling the tracking output automatically via a temporary file, so you would not need to manually write to or read back from `trajectory.csv`.  Let me know what you think and I can look into updating `NatsSimulationWrapper`.  Another alternative would be for you pass the output filename to your `simulation` method (with a default value of `None`), but that still would not take full advantage of the simulation wrapper.